core/src/main/java/org/apache/carbondata/core/scan/collector/impl/DictionaryBasedResultCollector.java

```diff
@@ -141,22 +141,50 @@ public DictionaryBasedResultCollector(BlockExecutionInfo blockExecutionInfos) {
       }
       fillMeasureData(scannedResult, row);
       if (scannedResult.complexParentIndexToQueryMap.toString().contains("StructQueryType")) {
```

Comment:
This check is not good in terms of performance,  every time generating a string from map and check contains is heavy operation. please simplify it. And also I don't think we require to check it for every row.

Reply:
yes, avoided string conversion.